### PR TITLE
add support for inplace integrand

### PR DESCRIPTION
**Pull request by kunyuan**

### CODE REVIEW

File: src/main.jl [hunk 1]
```diff
@@ -40,14 +42,18 @@
...
 
 # Examples
 ```julia-repl
-julia> integrate((x, c)->(x[1]^2+x[2]^2); var = Continuous(0.0, 1.0), dof = 2, print=-1)
-Integral 1 = 0.6668625385256122 ± 0.0009960142738129768   (chi2/dof = 1.07)
+integrate((x, c)->(x[1]^2+x[2]^2); var = Continuous(0.0, 1.0), dof = 2, print=-2, solver=:vegas)
+Integral 1 = 0.6663652080622751 ± 0.000490978424216832   (chi2/dof = 0.645)
+
+julia> integrate((x, f, c)-> (f=x[1]^2+x[2]^2); var = Continuous(0.0, 1.0), dof = 2, print=-2, solver=:vegas, inplace=true) # inplace version
+Integral 1 = 0.0 ± 3.3333333333333335e-11   (chi2/dof = 0.0)
```
Comment (dcerkoney) [inline src/main.jl:56]: The in-place integration result looks incorrect (c.f. line 53). Perhaps line 55 did not run correctly?

Comment (kunyuan) [reply src/main.jl:56]: Good catch, I fixed it with the new commit.

**Review by dcerkoney — APPROVED** [review]
Looks good to me!

File: src/main.jl [hunk 2]
```diff
@@ -24,7 +24,9 @@
 
  # Arguments
 
-- `integrand`:Function call to evaluate the integrand. It should accept an argument of the type [`Configuration`](@ref), and return a weight. 
+- `integrand`:Function call to evaluate the integrand.  
+              If inpace = false, then the signature of the integrand should be `integrand(var, config)`, where `var` is a vector of random variables, and `config` is the [`Configuration`](@ref) struct. It should return one or more weights, corresponding to the value of each component of the integrand for the given `var`.
```
Comment (iintSjds) [inline src/main.jl:28]: inpace -> inplace

**Pull request merged**